### DISCUSSION TITLE
JSON basic support for Float values

### DISCUSSION THREAD

**Discussion by jcontrerasf** [zephyrproject-rtos/zephyr#77665] [Ideas]
A few days ago, I needed to consume an API and realized that Zephyr's JSON utility doesn't support float parsing. It actually processes them as opaque values. That's why I decided to implement this feature myself, and I'd like to know if it's suitable/secure enough to be officially added to Zephyr.

This implementation is quite limited; it only supports one decimal place. However, in my opinion, this is much better than just returning 0.

The changes in `json.c` are as follows: (1) adding a separate `case JSON_TOK_FLOAT` in the `decode_value` function and (2) returning the correct `sizeof(float)` in `get_elem_size()`.

```
case JSON_TOK_FLOAT: {
  float *num = field;
  char *endptr;
  *num = strtol(value->start, &endptr, 10);
  if(*endptr == '.' && *(endptr + 1) >= '0' && *(endptr + 1) <= '9'){
    *num += (float)((*(endptr + 1) - '0') / 10.0);
  }else if (endptr != value->end) {
    return -EINVAL;
  }
  return 0;
}
```

**Comment by PeeJay** [comment]
Given that Zephyr is used on many IOT devices it seems crazy there is no json float support. If JSON_TOK_FLOAT is used the required strtod and friends are not available then just `#error` with a helpful error message about the situation.

I've been using ArduinoJson as I'm already familiar with it,

**Comment by jukkar** [comment]
If you have a patch for this, just create a PR to get feedback.

**Comment by ofirshe** [comment]
Following that discussion as I also need this feature. @jcontrerasf Are you going to submit a PR ?
